modules/xfem/src/userobjects/LevelSetCutUserObject.C

```diff
@@ -0,0 +1,157 @@
+                 " base element type is: ",
+                 libMesh::Utility::enum_to_string(elem->type()));
+
+    const Node * node1 = curr_side->get_node(0);
+    const Node * node2 = curr_side->get_node(1);
+
+    dof_id_type ls_dof_id_1 = node1->dof_number(_system.number(), _ls_var_number, 0);
+    dof_id_type ls_dof_id_2 = node2->dof_number(_system.number(), _ls_var_number, 0);
+
+    Number ls_node_1 = (*_solution)(ls_dof_id_1);
+    Number ls_node_2 = (*_solution)(ls_dof_id_2);
+
+    if (ls_node_1 * ls_node_2 < 0)
+    {
+      cut_elem = true;
+      Xfem::CutEdge mycut;
+      mycut._id1 = node1->id();
+      mycut._id2 = node2->id();
+      Real seg_int_frac = std::abs(ls_node_1) / std::abs(ls_node_1 - ls_node_2);
+      mycut._distance = seg_int_frac;
+      mycut._host_side_id = i;
+      cut_edges.push_back(mycut);
+    }
+  }
+
+  return cut_elem;
+}
+
+bool
+LevelSetCutUserObject::cutElementByGeometry(const Elem * elem,
+                                            std::vector<Xfem::CutFace> & cut_faces,
+                                            Real /*time*/) const
+{
+  bool cut_elem = false;
+
+  for (unsigned int i = 0; i < elem->n_sides(); ++i)
+  {
+    std::unique_ptr<Elem> curr_side = elem->side(i);
+    if (curr_side->dim() != 2)
+      mooseError("In LevelSetCutUserObject dimension of side must be 2, but it is ",
+                 curr_side->dim());
+    unsigned int n_edges = curr_side->n_sides();
+
+    std::vector<unsigned int> cut_edges;
+    std::vector<Real> cut_pos;
+
+    for (unsigned int j = 0; j < n_edges; j++)
+    {
+      // This returns the lowest-order type of side.
+      std::unique_ptr<Elem> curr_edge = curr_side->side(j);
+      if (curr_edge->type() != EDGE2)
+        mooseError("In LevelSetCutUserObject face edge must be EDGE2, but type is: ",
+                   libMesh::Utility::enum_to_string(curr_edge->type()),
+                   " base element type is: ",
+                   libMesh::Utility::enum_to_string(elem->type()));
+
+      const Node * node1 = curr_edge->get_node(0);
+      const Node * node2 = curr_edge->get_node(1);
+
+      dof_id_type ls_dof_id_1 = node1->dof_number(_system.number(), _ls_var_number, 0);
+      dof_id_type ls_dof_id_2 = node2->dof_number(_system.number(), _ls_var_number, 0);
+
+      Number ls_node_1 = (*_solution)(ls_dof_id_1);
+      Number ls_node_2 = (*_solution)(ls_dof_id_2);
+
+      if (ls_node_1 * ls_node_2 < 0)
+      {
+        Real seg_int_frac = std::abs(ls_node_1) / std::abs(ls_node_1 - ls_node_2);
+        cut_edges.push_back(j);
+        cut_pos.push_back(seg_int_frac);
+      }
+    }
+
+    if (cut_edges.size() == 2)
+    {
+      cut_elem = true;
+      Xfem::CutFace mycut;
+      mycut._face_id = i;
+      mycut._face_edge.push_back(cut_edges[0]);
+      mycut._face_edge.push_back(cut_edges[1]);
+      mycut._position.push_back(cut_pos[0]);
+      mycut._position.push_back(cut_pos[1]);
+      cut_faces.push_back(mycut);
+    }
```

Comment:
Could you add error messages for the cases when `cut_edges.size() != 2`?

Reply:
If the size !=2, then it will return false. No error message is needed here.